### PR TITLE
BUG: Ensure that we only cast if timezone is not None

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

**Comment by cpcloud** [comment]
Merging on green.

File: ibis/expr/api.py [hunk 1]
```diff
@@ -1884,7 +1884,9 @@ def between_time(arg, lower, upper, timezone=None):
     """
 
     if isinstance(arg.op(), _ops.Time):
-        arg = arg.op().args[0].cast(dt.Timestamp(timezone=timezone))
+        arg = arg.op().args[0]
```
Comment (jreback) [inline ibis/expr/api.py]: maybe a commento about why this is needed here?

Comment (cpcloud) [reply ibis/expr/api.py]: Done

**Comment by cpcloud** [comment]
Passing on circle: https://circleci.com/workflow-run/65990320-3cf3-4a5e-9a39-7ca17d43ec70
Passing on appveyor: https://ci.appveyor.com/project/cpcloud/ibis-xh5g1/build/1.0.602

Merging.

**Pull request closed without merging**